### ISSUE TITLE
[spine-ts] Not able to build spine-webgl.js 

### ISSUE THREAD

**Issue by jonchun** [EsotericSoftware/spine-runtimes#1502]
Getting the following while trying to build the webGL version.

```
~/Documents/Projects/spine-runtimes/spine-ts  2  tsc -p tsconfig.json
webgl/src/WebGL.ts:55:5 - error TS2322: Type 'HTMLCanvasElement | OffscreenCanvas' is not assignable to type 'HTMLCanvasElement'.
  Type 'OffscreenCanvas' is missing the following properties from type 'HTMLCanvasElement': toBlob, toDataURL, transferControlToOffscreen, accessKey, and 237 more.

55  			this.canvas = this.gl.canvas;
    			~~~~~~~~~~~


Found 1 error.
```

I've patched line 32 on `webgl/src/WebGL.ts` to
```public canvas: HTMLCanvasElement | OffscreenCanvas;``` 
to get it to build, but I don't know if there are additional implications to this.

Type: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/canvas

If anyone who knows this project better than me could chime in and confirm whether my change is safe, I would be happy to make a simple PR.

**Comment by jonchun** [comment]
I was also trying to build 3.8, but the code doesn't look to be any different in 3.9-beta. Are my tsc settings somehow messed up?

**Comment by badlogic** [comment]
Where does the dependency to `OffscreenCanvas` come from? That's not part of our source tree here, neither on the 3.8 nor the 3.9-beta branch.

https://github.com/EsotericSoftware/spine-runtimes/blob/3.8/spine-ts/webgl/src/WebGL.ts#L55

**Comment by jonchun** [comment]
> Where does the dependency to `OffscreenCanvas` come from? That's not part of our source tree here, neither on the 3.8 nor the 3.9-beta branch.
> 
> https://github.com/EsotericSoftware/spine-runtimes/blob/3.8/spine-ts/webgl/src/WebGL.ts#L55

as per https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/canvas
> The WebGLRenderingContext.canvas property is a read-only reference to the HTMLCanvasElement or OffscreenCanvas object that is associated with the context. It might be null if it is not associated with a <canvas> element or an OffscreenCanvas object.

In line 55 that you linked, I'm seeing that a reference to `this.gl.canvas` which is of type `WebGLRenderingContext`. I'm not super familiar with canvas types and am really new to the TypeScript ecosystem so I could be looking at the wrong documentation.

Update: I see now that [OffscreenCanvas](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) is experimental. Is there something that needs to be in `tsconfig.json` to avoid experimental types?

**Comment by badlogic** [comment]
What TypeScript compiler version are you using?

**Comment by jonchun** [comment]
On MacOS 10.14.6
```
$ tsc -V
Version 3.6.3
```

**Comment by badlogic** [comment]
Thanks for reporting! I was on an older TypeScript compiler version and didn't catch this. I have fixed everything up on the 3.8 and 3.9-beta branches.
